src/modules/Profile/lib/Profile/Api/User.php

```diff
@@ -640,4 +640,143 @@ public function encodeurl($args)
         // construct the custom url part
         return $args['modname'] . '/' . $args['func'] . '/' . $vars;
     }
+
+    /**
+     * Get module links
+     *
+     * @return array array of links
+     */
+    public function getLinks()
+    {
+        if (!SecurityUtil::checkPermission('Profile::', '::', ACCESS_OVERVIEW)) {
```

Comment:
This permission is onli of to see the menu. below depending on if user is logged in, are performed other checks.
Have in mind, that this is mainly transferring from existing permission checks in the profilemodulelinks plugin.